### PR TITLE
Dependencies step - Raise exception after logging

### PR DESCRIPTION
**Pull request by GalRabin**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/31616

## Description
Raising exception when one of the pool processes raise an exception.

Example for failing build - https://app.circleci.com/pipelines/github/demisto/content/52039/workflows/df4eb01d-7e9d-4985-afe6-f5e17dad1657/jobs/219761

## Screenshots
image

### CODE REVIEW

File: Tests/Marketplace/packs_dependencies.py [hunk 1]
```diff
@@ -27,6 +30,50 @@ def option_handler():
...
+    Yields:
+        ProcessPool: Pebble process pool.
+    """
+    with ProcessPool(max_workers=3) as pool:
+        try:
+            yield pool
+        except KeyboardInterrupt:
+            logging.info("\nCTRL+C Pressed!\nGracefully release all resources due to keyboard interrupt...")
+            pool.stop()
+            pool.join()
```
Comment (yaakovi) [inline Tests/Marketplace/packs_dependencies.py:46]: move to `finally` scope instead of duplicating in each scope

Comment (hod-alpert) [inline Tests/Marketplace/packs_dependencies.py:46]: Move the pool.stop, pool.join to a finally clause in all cases

File: Tests/Marketplace/packs_dependencies.py [hunk 2]
```diff
@@ -27,6 +30,50 @@ def option_handler():
...
+def ProcessPoolHandler() -> ProcessPool:
+    """ Process pool Handler which terminate all processes in case of Exception.
+
+    Yields:
+        ProcessPool: Pebble process pool.
+    """
+    with ProcessPool(max_workers=3) as pool:
+        try:
+            yield pool
+        except KeyboardInterrupt:
```
Comment (hod-alpert) [inline Tests/Marketplace/packs_dependencies.py:43]: Why is this case needed?
this is not sdk and the behavior is the same as every exception

**Review by hod-alpert — APPROVED** [review]

File: Tests/Marketplace/packs_dependencies.py (hunk 2 above)
Comment (GalRabin) [reply Tests/Marketplace/packs_dependencies.py]: Done, Correct

File: Tests/Marketplace/packs_dependencies.py [hunk 3]
```diff
@@ -27,6 +30,43 @@ def option_handler():
...
+
+    Yields:
+        ProcessPool: Pebble process pool.
+    """
+    with ProcessPool(max_workers=3) as pool:
+        try:
+            yield pool
+        except Exception as e:
+            logging.exception(e)
+            logging.error("Gracefully release all resources due to Error...")
```
Comment (hod-alpert) [inline Tests/Marketplace/packs_dependencies.py:45]:
```suggestion
        except Exception:
            logging.exception("Gracefully release all resources due to Error...")
```

**Review by yaakovi — APPROVED** [review]

**Pull request merged**